extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  @Override
+  public CloseableIterator<InputRow> read()
+  {
+    return CloseableIterators.withEmptyBaggage(readAsList().iterator());
+  }
+
+  List<InputRow> readAsList()
+  {
+    try {
+      return parseMetricsData(MetricsData.parseFrom(source.getBuffer()));
+    }
+    catch (InvalidProtocolBufferException e) {
+      throw new ParseException(e, "Protobuf message could not be parsed");
+    }
+  }
+
+  private List<InputRow> parseMetricsData(final MetricsData metricsData)
+  {
+    return metricsData.getResourceMetricsList()
+        .stream()
+        .flatMap(resourceMetrics -> {
+          Map<String, Object> resourceAttributes = resourceMetrics.getResource()
+              .getAttributesList()
+              .stream()
+              .collect(Collectors.toMap(kv -> resourceAttributePrefix + kv.getKey(), kv -> kv.getValue().getStringValue()));
+          return resourceMetrics.getInstrumentationLibraryMetricsList()
+              .stream()
+              .flatMap(libraryMetrics -> libraryMetrics.getMetricsList()
+                  .stream()
+                  .flatMap(metric -> parseMetric(metric, resourceAttributes).stream()));
+        })
+        .collect(Collectors.toList());
+  }
+
+  private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAttributes)
+  {
+    List<InputRow> inputRows;
+    String metricName = metric.getName();
+    switch (metric.getDataCase()) {
+      case INT_SUM: {
+        inputRows = new ArrayList<>(metric.getIntSum().getDataPointsCount());
+        metric.getIntSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseIntDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case INT_GAUGE: {
+        inputRows = new ArrayList<>(metric.getIntGauge().getDataPointsCount());
+        metric.getIntGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseIntDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case SUM: {
+        inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
+        metric.getSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case GAUGE: {
+        inputRows = new ArrayList<>(metric.getGauge().getDataPointsCount());
+        metric.getGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      // TODO Support and SUMMARY metrics
+      default:
+        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
+    }
+    return inputRows;
+  }
+
+  private InputRow parseIntDataPoint(IntDataPoint dataPoint, Map<String, Object> resourceAttributes, String metricName)
```

Comment:
we talked about adding a comment here to explain that we are ok with the performance impact of copying the data here since this is only for backwards compatibility with deprecated formats.